src/modules/NetworkController/public/Get-SdnServiceFabricPartition.ps1

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+function Get-SdnServiceFabricPartition {
+    <#
+    .SYNOPSIS
+        Gets information about the partitions of a specified Service Fabric partition or service from Network Controller.
+    .PARAMETER ApplicationName
+        A service fabric application name that exists on the provided ring, such as fabric:/NetworkController.
+    .PARAMETER ServiceName
+        A service fabric service name that is under the provided ApplicationName on the provided ring, such as fabric:/NetworkController/ApiService.
+    .PARAMETER ServiceTypeName
+        A service fabric service TypeName, such as VSwitchService.
+    .PARAMETER NetworkController
+        Specifies the name of the network controller node on which this cmdlet operates.
+    .PARAMETER Credential
+        Specifies a user account that has permission to perform this action. The default is the current user.
+    .EXAMPLE
+        PS> Get-SdnServiceFabricPartition -PartitionId 1a7a780e-dbfe-46d3-92fb-76908a95ce54
+    .EXAMPLE
+        PS> Get-SdnServiceFabricPartition -NetworkController 'Prefix-NC01' -Credential (Get-Credential) -ServiceTypeName 'ApiService'
+    .EXAMPLE
+        PS> Get-SdnServiceFabricPartition -NetworkController 'Prefix-NC01' -Credential (Get-Credential) -ServiceName 'fabric:/NetworkController/ApiService'
+    #>
+
+    [CmdletBinding(DefaultParameterSetName = 'NamedService')]
+    param(
+        [Parameter(Mandatory = $false, ValueFromPipeline = $false, ParameterSetName = 'NamedService')]
+        [Parameter(Mandatory = $false, ValueFromPipeline = $false, ParameterSetName = 'NamedServiceTypeName')]
+        [System.String]$ApplicationName = 'fabric:/NetworkController',
+
+        [Parameter(Mandatory = $true, ValueFromPipeline = $false, ParameterSetName = 'NamedService')]
+        [System.String]$ServiceName,
+
+        [Parameter(Mandatory = $false, ValueFromPipeline = $false, ParameterSetName = 'NamedServiceTypeName')]
```

Comment:
Should this be Mandatory?

Reply:
@luyaof works in either instance since won't accept empty value and defaults to NamedService if no params specified. Pushed changes though to set mandatory to $true